lib/core/api.js

```diff
@@ -166,8 +166,21 @@ api.user.updateProfile = function(userId, updates, cb){
     if(err){
       return cb(err);
     }
+    var currentEmail = _.get(profile, 'emails[0]');
     var newProfile = actionUtils.mergeProfileUpdates(profile, updates);
-    return api.user.addProfile(userId, newProfile, cb);
+    var emails = _.get(updates, 'emails');
+    if (_.isArray(emails) && emails.length === 1 && emails[0] !== currentEmail){
```

Comment:
do you think its worth pulling this out into its own function? just thinking that its kinda hard to tell what is happening here at a glance and maybe being in own function it would be a little clearer,  or having a brief comment as to whats going on. Thoughts?